interfaces/compatibility/compatibility_test.go

```diff
@@ -0,0 +1,256 @@
+		{"foo-(1..3)-4", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+			{Tag: "foo", Values: []compatibility.CompatRange{{1, 3}, {4, 4}}}}}},
+		{"foo3e4-4-(1..3)-bar-2025-8-2", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+			{Tag: "foo3e4", Values: []compatibility.CompatRange{{4, 4}, {1, 3}}},
+			{Tag: "bar", Values: []compatibility.CompatRange{{2025, 2025}, {8, 8}, {2, 2}}},
+		}}},
+		{"foo3e4-4-(1..3)-bar-2025-8-2-libxxlongname-56-(4..90)", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+			{Tag: "foo3e4", Values: []compatibility.CompatRange{{4, 4}, {1, 3}}},
+			{Tag: "bar", Values: []compatibility.CompatRange{{2025, 2025}, {8, 8}, {2, 2}}},
+			{Tag: "libxxlongname", Values: []compatibility.CompatRange{{56, 56}, {4, 90}}},
+		}}},
+	} {
+		c.Logf("tc %d: %+v", i, tc)
+
+		cfield, err := compatibility.DecodeCompatField(tc.compat, nil)
+		c.Assert(err, IsNil)
+		c.Check(cfield, DeepEquals, tc.cfield)
+	}
+}
+
+func (s *CompatSuite) TestValidCompatFieldsWithSpec(c *C) {
+	for i, tc := range []struct {
+		compat string
+		cfield *compatibility.CompatField
+		spec   *compatibility.CompatSpec
+	}{
+		{"foo",
+			&compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{0, 0}}}}},
+			&compatibility.CompatSpec{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{0, math.MaxUint}}}}},
+		},
+		{"foo-2",
+			&compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{2, 2}}}}},
+			&compatibility.CompatSpec{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{0, 3}}}}},
+		},
+		{"foo-(2..6)",
+			&compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{2, 6}}}}},
+			&compatibility.CompatSpec{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{2, 6}}}}},
+		},
+		{"foo-(2..6)",
+			&compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{2, 6}}}}},
+			&compatibility.CompatSpec{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{1, 10}}}}},
+		},
+		{"foo-(2..6)-bar-7",
+			&compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{2, 6}}},
+				{Tag: "bar", Values: []compatibility.CompatRange{{7, 7}}},
+			}},
+			&compatibility.CompatSpec{Dimensions: []compatibility.CompatDimension{
+				{Tag: "foo", Values: []compatibility.CompatRange{{1, 10}}},
+				{Tag: "bar", Values: []compatibility.CompatRange{{7, 10}}},
+			}},
+		},
+	} {
+		c.Logf("tc %d: %+v", i, tc)
+
+		cfield, err := compatibility.DecodeCompatField(tc.compat, nil)
```

Comment:
The test isn't passing the spec
```suggestion
		cfield, err := compatibility.DecodeCompatField(tc.compat, tc.spec)
```